base/linalg/triangular.jl

```diff
@@ -889,6 +889,201 @@ for (f, g) in ((:/, :A_rdiv_B!), (:A_rdiv_Bc, :A_rdiv_Bc!), (:A_rdiv_Bt, :A_rdiv
+    if alpha2 <= theta[2]
+        m = alpha2<=theta[1]?1:2
+        foundm = true
+    end
+
+    while ~foundm
+        more = false
+        if s > s0
+            d3 = norm(AmI^3, 1)^(1/3)
+        end
+        d4 = norm(AmI^4, 1)^(1/4)
+        alpha3 = max(d3, d4)
+        if alpha3 <= theta[tmax]
+            for j = 3:tmax
+                if alpha3 <= theta[j]
+                    break
+                end
+            end
+            if j <= 6
+                m = j
+                break
+            elseif alpha3/2 <= theta[5] && p < 2
+                more = true
+                p = p + 1
+           end
+        end
+
+        if ~more
+            d5 = norm(AmI^5, 1)^(1/5)
+            alpha4 = max(d4, d5);
+            eta = min(alpha3, alpha4);
+            if eta <= theta[tmax]
+                j = 0
+                for j = 6:tmax
+                    if eta <= theta[j]
+                        m = j
+                        break
+                    end
+                end
+                break
+            end
+        end
+
+        if s == maxsqrt
+            m = tmax
+            break
+        end
+        A = sqrtm(A)
+        AmI = A - UpperTriangular(eye(n))
+        s = s + 1
+    end
+
+    # Compute accurate superdiagonal of T
+    p = (1/2^s)
+    for k = 1:n-1
+        Ak = A0[k,k]
+        Akp1 = A0[k+1,k+1]
+        Akp = Ak^p
+        Akp1p = Akp1^p
+        A[k,k] = Akp
+        A[k+1,k+1] = Akp1p
+        if Ak == Akp1
+            A[k,k+1] = p * A0[k,k+1] * Ak^(p-1)
+        elseif abs(Ak) < abs(Akp1)/2 || abs(Akp1) < abs(Ak)/2
+            A[k,k+1] = A0[k,k+1] * (Akp1p - Akp) / (Akp1 - Ak)
+        else
+            logAk = log(Ak)
+            logAkp1 = log(Akp1)
+            w = atanh((Akp1 - Ak)/(Akp1 + Ak)) + im*pi*ceil((imag(logAkp1-logAk)-pi)/(2*pi))
+            dd = 2 * exp(p*(logAk+logAkp1)/2) * sinh(p*w) / (Akp1 - Ak);
+            A[k,k+1] = A0[k,k+1] * dd
+        end
+    end
+
+    # Compute accurate diagonal of T
+    for i = 1:n
+        a = A0[i,i]
+        if s == 0
+            r = a - 1
+        end
+        s0 = s
+        if angle(a) >= pi/2
+            a = sqrt(a)
+            s0 = s - 1
+        end
+        z0 = a - 1
+        a = sqrt(a)
+        r = 1 + a
+        for j = 1:s0-1
+            a = sqrt(a)
+            r = r * (1 + a)
+        end
+        A[i,i] = z0 / r
+    end
+
+    # Compute the Gauss-Legendre quadrature formula
+    R = zeros(Float64, m, m)
+    for i = 1:m - 1
+        R[i,i+1] = i / sqrt((2 * i).^2-1)
+        R[i+1,i] = R[i,i+1]
+    end
+    x,V = eig(R)
+    w = Array(Float64, m)
+    for i = 1:m
+        x[i] = (x[i] + 1) / 2
+        w[i] = (V[1,i])^2
+    end
+
+    # Compute the Padé approximation
+    B = zeros(T, n, n)
+    Y = zeros(T, n, n)
+    for k = 1:m
+        for j = 1:n
+            B[j,j] = 1. + A[j,j] * x[k]
+            for i = 1:j-1
+                B[i,j] = A[i,j] * x[k]
+            end
+        end
+        B = A * inv(B)
```

Comment:
There quite a few temporaries allocated in this loop so eventually we might want to update `B` in place.